pkg/virt-launcher/virtwrap/converter/converter.go

```diff
@@ -1482,108 +1457,95 @@ func Convert_v1_VirtualMachine_To_api_Domain(vmi *v1.VirtualMachineInstance, dom
+			// https://libvirt.org/formatdomain.html#elementsNICSEthernet
+			domainIface.Type = "ethernet"
+			if iface.BootOrder != nil {
+				domainIface.BootOrder = &api.BootOrder{Order: *iface.BootOrder}
+			} else {
+				domainIface.Rom = &api.Rom{Enabled: "no"}
 			}
+		} else if iface.Slirp != nil {
+			domainIface.Type = "user"
 
-			if iface.Bridge != nil || iface.Masquerade != nil {
-				// TODO:(ihar) consider abstracting interface type conversion /
-				// detection into drivers
-
-				// use "ethernet" interface type, since we're using pre-configured tap devices
-				// https://libvirt.org/formatdomain.html#elementsNICSEthernet
-				domainIface.Type = "ethernet"
-				if iface.BootOrder != nil {
-					domainIface.BootOrder = &api.BootOrder{Order: *iface.BootOrder}
-				} else {
-					domainIface.Rom = &api.Rom{Enabled: "no"}
-				}
-			} else if iface.Slirp != nil {
-				domainIface.Type = "user"
-
-				// Create network interface
-				if domain.Spec.QEMUCmd == nil {
-					domain.Spec.QEMUCmd = &api.Commandline{}
-				}
+			// Create network interface
+			if domain.Spec.QEMUCmd == nil {
+				domain.Spec.QEMUCmd = &api.Commandline{}
+			}
 
-				if domain.Spec.QEMUCmd.QEMUArg == nil {
-					domain.Spec.QEMUCmd.QEMUArg = make([]api.Arg, 0)
-				}
+			if domain.Spec.QEMUCmd.QEMUArg == nil {
+				domain.Spec.QEMUCmd.QEMUArg = make([]api.Arg, 0)
+			}
 
-				// TODO: (seba) Need to change this if multiple interface can be connected to the same network
-				// append the ports from all the interfaces connected to the same network
-				err := createSlirpNetwork(iface, *net, domain)
-				if err != nil {
-					return err
-				}
-			} else if iface.Macvtap != nil {
-				if net.Multus == nil {
-					return fmt.Errorf("macvtap interface %s requires Multus meta-cni", iface.Name)
-				}
+			// TODO: (seba) Need to change this if multiple interface can be connected to the same network
+			// append the ports from all the interfaces connected to the same network
+			err := createSlirpNetwork(iface, *net, domain)
+			if err != nil {
+				return err
+			}
+		} else if iface.Macvtap != nil {
+			if net.Multus == nil {
+				return fmt.Errorf("macvtap interface %s requires Multus meta-cni", iface.Name)
+			}
 
-				domainIface.Type = "ethernet"
-				if iface.BootOrder != nil {
-					domainIface.BootOrder = &api.BootOrder{Order: *iface.BootOrder}
-				} else {
-					domainIface.Rom = &api.Rom{Enabled: "no"}
-				}
+			domainIface.Type = "ethernet"
+			if iface.BootOrder != nil {
+				domainIface.BootOrder = &api.BootOrder{Order: *iface.BootOrder}
+			} else {
+				domainIface.Rom = &api.Rom{Enabled: "no"}
 			}
-			domain.Spec.Devices.Interfaces = append(domain.Spec.Devices.Interfaces, domainIface)
 		}
+		domain.Spec.Devices.Interfaces = append(domain.Spec.Devices.Interfaces, domainIface)
 	}
 
+	domain.Spec.Devices.HostDevices = append(domain.Spec.Devices.HostDevices, c.SRIOVDevices...)
```

Comment:
I think this line should be moved to `Convert_HostDevices_And_GPU`
What you think?